utils/link_utils.js

```diff
@@ -0,0 +1,66 @@
+const axios = require('axios');
+const { ytApiId } = require('../config.json');
+
+// regex list with all compatible
+const sourceRegexList = {
+    YouTube: /^(?:https?:\/\/)?(?:www\.)?(?:youtube\.com\/watch\?v=|youtu\.be\/)([a-zA-Z0-9_-]+)/,
+    YouTubePlaylist: /^(?:https?:\/\/)?(?:www\.)?youtube\.com\/playlist\?list=([a-zA-Z0-9_-]+)/,
+};
+
+// matches url against regex
+function identifyUrlType(url) {
+    for (const source in sourceRegexList) {
+        const regex = sourceRegexList[source];
+        const match = url.match(regex);
+        if (match) {
+            return { source, id: match[1] };
```

Comment:
can we have here return { source: source, id: match[1] };  to keep the standard from line 19